### ISSUE TITLE
Do we have json api to write tags ? or data points

### ISSUE THREAD

**Issue by abhishekricky** [SCADA-LTS/Scada-LTS#484]
Do we have json api to write tags ? or data points

**Comment by grzesiekb** [comment]
Yes, We have REST API to save and read data for data point. I will try to prepare an example of the REST API. It is to improve the security of the REST API.

**Comment by abhishekricky** [comment]
I just browsed over API, did find following api

"/api/point_value/setValue/{xid}/{type}/{value}"

Value does change on the watchlist point, but it doesnt change on modbus device and later it reverts to original

**Comment by abhishekricky** [comment]
@grzesiekb  I tried the SOAP api for writeData but getting below error 

<?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <soapenv:Body>
        <soapenv:Fault>
            <faultcode>soapenv:Server.userException</faultcode>
            <faultstring>org.xml.sax.SAXException: No deserializer for {http://www.w3.org/2001/XMLSchema}anyType</faultstring>
            <detail>
                <ns1:hostname xmlns:ns1="http://xml.apache.org/axis/">sularemote</ns1:hostname>
            </detail>
        </soapenv:Fault>
    </soapenv:Body>
</soapenv:Envelope>

**Comment by grzesiekb** [comment]
@abhishekricky Could you add the definition of modbus which you didn't update from REST API.

**Comment by abhishekricky** [comment]
api end point I used -> /api/point_value/setValue/{xid}/{type}/{value}

/api/point_value/setValue/DP_152354/2/16 

And it was Holding Register -> Address 34 ( Format is unsigned Integer )

**Comment by grzesiekb** [comment]
Please write to me email gbylica@softq.pl.

**Comment by rinose** [comment]
When I use configureDataSource wsdl soap api I have same issue 
org.xml.sax.SAXException: No deserializer for {http://www.w3.org/2001/XMLSchema}anyType

Have you solved?

**Comment by Patrykb0802** [comment]
To edit values of your data points use writeStringData end point. For example:

![Image](https://github.com/user-attachments/assets/64b5dd73-7a3e-46f1-9889-d2192cd54b2f)

vo:itemName = name of your data point
vo:dataType = Java data type not Scada data type, DOUBLE will do in most cases
vo:value = value you want your data point to be set at

For quality and timestamp you can use readData and check what are the parameters of point you want to modify:

![Image](https://github.com/user-attachments/assets/01903455-05d8-48b8-b895-122ec7df1f17)

For REST API I also checked setValue and it;s working flawlessly. Usage example:

![Image](https://github.com/user-attachments/assets/10a22303-1ac8-41e2-8ac5-3da36101df60)
/api/point_value/setValue/{xid}/{type}/{value}

{type} = 0 - unknown, 1 - binary, 2 - multistate, 3 - double, 4 - string
